lib/reek/smells/attribute.rb

```diff
@@ -12,10 +12,20 @@ module Smells
     # +attr_reader+, +attr_writer+ and +attr_accessor+ -- including those
     # that are private.
     #
-    # TODO: Eliminate private attributes
     # TODO: Catch attributes declared "by hand"
     #
     class Attribute < SmellDetector
+      ATTR_DEFN_METHODS = [:attr, :attr_reader, :attr_writer, :attr_accessor]
+      VISIBILITY_MODIFIERS = [:private, :public, :protected]
+      VISIBLITY_MODE = :public
+
+      def initialize(*args)
+        @visiblity_tracker = {}
+        @visiblity_mode = :public
```

Comment:
You have a constant VISIBILITY_MODE and an ivar "@visibility_mode". That's highly confusing. Can't we get rid of the ivar?